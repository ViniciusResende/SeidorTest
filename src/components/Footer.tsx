import React, { useContext } from 'react'
import { Wrapper } from '../styles/Footer.styles';

import CircleIcon from '../assets/circle.svg';
import { AppContext } from '../contexts/AppContext';

const Footer: React.FC = () => {
  const {
    menuListHandler,
    menuRegisterHandler,
  } = useContext(AppContext);

    return (
    <Wrapper>  
      <div className="footerContainer">
        <div className="navigationContainer">
          <button onClick={menuListHandler}>Lista</button>
          <button onClick={menuRegisterHandler}>Cadastro</button>
        </div>
        <div className="sideContainer">
          <img src={CircleIcon} alt="logo"/>
          <p>Copyright © 2021.</p>
          <p>Todos os direitos reservados à</p>
          <p>Vinícius Alves</p>
        </div>
      </div>
    </Wrapper>
  )
}

export default Footer