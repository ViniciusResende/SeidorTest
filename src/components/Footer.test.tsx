import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Footer from './Footer';

const NewFooter = () => {
  return (
    <BrowserRouter>
      <Footer />
    </BrowserRouter>
  )
}

test('Should render the Footer with navigation anchors', () => {
  const tree = renderer
    .create(<NewFooter />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});