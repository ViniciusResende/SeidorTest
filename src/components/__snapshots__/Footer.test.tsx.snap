// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Footer with navigation anchors 1`] = `
<div
  className="sc-bdfBwQ dzHHMX"
>
  <div>
    <a
      className="sc-dlfnbm kJyqox"
      href="/"
      onClick={[Function]}
    >
      Lista
    </a>
    <a
      className="sc-dlfnbm kJyqox"
      href="/register"
      onClick={[Function]}
    >
      Cadastro
    </a>
    <a
      className="sc-dlfnbm kJyqox"
      href="/update"
      onClick={[Function]}
    >
      Atualizar
    </a>
  </div>
  <div
    className="sc-gsTCUz jeMrqT"
  >
    <img
      alt="logo"
      src="circle.svg"
    />
    <p>
      Copyright © 2021.
    </p>
    <p>
      Todos os direitos reservados à
    </p>
    <p>
      Vinícius Alves
    </p>
  </div>
</div>
`;
